src/index.ts

```diff
@@ -86,6 +87,10 @@ function run(updateDetails?: IUpdateDetails) {
         console.error(cerr`--production and --onlypre cannot be enabled simultaneously`);
         process.exit(1);
     }
+    const mainRepoPath = AssetsCompiler.getMainRepoPath(cli.flags.localOnly);
+    if (mainRepoPath && (path.basename(mainRepoPath) !== 'main') && (!cli.flags.onlypre)) {
```

Comment:
If we run in `localOnly` we also don't care about the name since in that case we expect the caller to know where he's looking. So we should also skip the warning in that case :)